google-datacatalog-qlik-connector/src/google/datacatalog_connectors/qlik/scrape/repository_services_api_helper.py

```diff
@@ -63,7 +63,7 @@ def get_full_stream_list(self, session):
         url = f'{self.__base_api_endpoint}' \
               f'/stream/full?Xrfkey={constants.XRFKEY}'
 
-        return session.get(url=url, headers=self.__headers).json()
+        return session.get(url=url, headers=self.__common_headers).json()
 
     def get_windows_authentication_url(self, session):
```

Comment:
Question about the design of `repository_services_api_helper.py` and `engine_api_helper`.

Those classes act as facades on top of different Qlik API's.

They expose the `get_windows_authentication_url` and have the authentication responsibility done by the scraper class.

Shouldn't each method: `__initialize_qrs_api_session` and `__initialize_engine_api_auth_cookie` belong to each class?

- `__initialize_qrs_api_session` to the `RepositoryServicesAPIHelper ` class.
- `__initialize_engine_api_auth_cookie`  to the `EngineAPIHelper` class.

Those classes seem to be facade classes but they have the authentication coupled with the scraper class. Let me known if this makes sense.

Reply:
Nice catch, @mesmacosta! I needed to refactor the code and tests to address this, but I enjoyed the proposed design. :).